### PR TITLE
[Merged by Bors] - Organized scene_viewer into plugins for reuse and organization

### PR DESCRIPTION
**Pull request by zardini123**
# Objective

This PR reorganizes majority of the scene viewer example into a module of plugins which then allows reuse of functionality among new or existing examples.  In addition, this enables the scene viewer to be more succinct and showcase the distinct cases of camera control and scene control.

This work is to support future work in organization and future examples.  A more complicated 3D scene example has been requested by the community (#6551) which requests functionality currently included in scene_viewer, but previously inaccessible.  The future example can now just utilize the two plugins created here.  The existing example [animated_fox example] can utilize the scene creation and animation control functionality of `SceneViewerPlugin`.

## Solution

- Created a `scene_viewer` module inside the `tools` example folder.
- Created two plugins:  `SceneViewerPlugin` (gltf scene loading, animation control, camera tracking control, light control) and `CameraControllerPlugin` (controllable camera).
- Original `scene_viewer.rs` moved to `scene_viewer/main.rs` and now utilizes the two plugins.

### CODE REVIEW

File: examples/tools/scene_viewer/camera_controller_plugin.rs [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+            friction: 0.5,
+            pitch: 0.0,
+            yaw: 0.0,
+            velocity: Vec3::ZERO,
+        }
+    }
+}
+
+#[derive(Default)]
+pub struct CameraControllerPlugin;
```
Comment (alice-i-cecile) [inline examples/tools/scene_viewer/camera_controller_plugin.rs:53]: Doc comments either here or at the module level would be nice.

File: examples/tools/scene_viewer/main.rs [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+use camera_controller_plugin::{CameraController, CameraControllerPlugin};
+use scene_viewer_plugin::{SceneHandle, SceneViewerPlugin};
+
+#[derive(Debug, Hash, PartialEq, Eq, Clone, SystemLabel)]
+struct CameraControllerCheckSystem;
+
+fn main() {
+    println!(
+        "
+Controls:
```
Comment (alice-i-cecile) [inline examples/tools/scene_viewer/main.rs:26]: IMO this should be done via a `Display` impl on the `CameraController` struct.

File: examples/tools/scene_viewer/mod.rs [hunk 3]
```diff
@@ -0,0 +1,5 @@
+mod camera_controller_plugin;
+mod scene_viewer_plugin;
+
+pub use camera_controller_plugin::*;
```
Comment (alice-i-cecile) [inline examples/tools/scene_viewer/mod.rs:4]: I don't like these `pub use *` invocations here. I don't think they make the structure clearer. I would rather just make the modules pub.

File: examples/tools/scene_viewer/main.rs [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+};
+
+mod camera_controller_plugin;
+mod scene_viewer_plugin;
+
+use camera_controller_plugin::{CameraController, CameraControllerPlugin};
+use scene_viewer_plugin::{SceneHandle, SceneViewerPlugin};
+
+#[derive(Debug, Hash, PartialEq, Eq, Clone, SystemLabel)]
+struct CameraControllerCheckSystem;
```
Comment (alice-i-cecile) [inline examples/tools/scene_viewer/main.rs:21]: We can probably eliminate this type: typed system labels rarely make sense when they're non-pub.

File: examples/tools/scene_viewer/main.rs [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+                },
+                ..default()
+            })
+            .set(AssetPlugin {
+                asset_folder: std::env::var("CARGO_MANIFEST_DIR")
+                    .unwrap_or_else(|_| ".".to_string()),
+                watch_for_changes: true,
+            }),
+    )
+    .add_plugin(CameraControllerPlugin::default())
```
Comment (alice-i-cecile) [inline examples/tools/scene_viewer/main.rs:64]: These plugins shouldn't have `Default` impls: instead, just initialize them as bare zero-field structs :)

**Review by alice-i-cecile — CHANGES_REQUESTED** [review]
I really like the direction of this. Left some feedback on code quality :)

File: examples/tools/scene_viewer/main.rs (hunk 2 above)
Comment (zardini123) [reply examples/tools/scene_viewer/main.rs]: Moved both control information to respective plugins to a `Display` impl

**Comment by zardini123** [comment]
Hi @alice-i-cecile!  Thank you very much for the review!  I have updated my PR to reflect your feedback.  Does these new changes match your expectation?

File: examples/tools/scene_viewer/camera_controller_plugin.rs (hunk 1 above)
Comment (zardini123) [reply examples/tools/scene_viewer/camera_controller_plugin.rs:84]: Created doc comments for both plugins

**Comment by zardini123** [comment]
I have updated the PR to match upstream and fixed some Ci issues.

File: examples/tools/scene_viewer/camera_controller_plugin.rs [hunk 6]
```diff
@@ -0,0 +1,174 @@
+//! A freecam-style camera controller plugin.
+//! To use:
+//! - Load `CameraControllerPlugin` into your App
```
Comment (alice-i-cecile) [inline examples/tools/scene_viewer/camera_controller_plugin.rs:3]:
```suggestion
//! - Copy the code for the `CameraControllerPlugin` and add it to your App
```

**Review by alice-i-cecile — APPROVED** [review]
Awesome! This is a great step forward on both cleaning up the example and letting other people easily reuse this code.

File: examples/tools/scene_viewer/camera_controller_plugin.rs (hunk 6 above)
Comment (zardini123) [reply examples/tools/scene_viewer/camera_controller_plugin.rs]: Modified both plugin doc comments to be more explicit regarding common use case of end-users, as per suggestion.

**Review by hymm — APPROVED** [review]
It's nice to have an example that is broken up into multiple files.

**Pull request closed without merging**